pkg/controller/application/rolloutblock.go

```diff
@@ -0,0 +1,54 @@
+package application
+
+import (
+	"fmt"
+	corev1 "k8s.io/api/core/v1"
+
+	shipper "github.com/bookingcom/shipper/pkg/apis/shipper/v1alpha1"
+	apputil "github.com/bookingcom/shipper/pkg/util/application"
+	"github.com/bookingcom/shipper/pkg/util/rolloutblock"
+)
+
+func (c *Controller) processRolloutBlocks(app *shipper.Application, nsRBs, gbRBs []*shipper.RolloutBlock) bool {
+	appOverrideRBs := rolloutblock.NewOverride(app.GetAnnotations()[shipper.RolloutBlocksOverrideAnnotation])
+	rbs := append(nsRBs, gbRBs...)
+
+	existingRBs := rolloutblock.NewOverrideFromRolloutBlocks(rbs)
+	obsoleteRbs := appOverrideRBs.Diff(existingRBs)
+
+	if len(obsoleteRbs) > 0 {
+		for o := range obsoleteRbs {
+			c.removeRolloutBlockFromAnnotations(appOverrideRBs, o, app)
```

Comment:
Proposed todo: move it to rolloutblockutils

Reply:
Can it be done in rolloutblockutil? It is updating the annotation of an application object